### ISSUE TITLE
Branching strategy

### ISSUE THREAD

**Issue by Aaronontheweb** [akkadotnet/akka.net#97]
**TL;DR;** I thought it would be easier for people to contribute if I added a layer of indirection from the stable master and am seeking additional input from fine developers like yourself.

Since Akka.NET is still in an early phase and there's lots of changes to be done, I went ahead and created a new `dev` branch of the project.

The `master` branch will reflect the contents of the most recent published AkkaDotNet Nuget package, whereas the `dev` branch might be unstable and contain yet-to-be-published changes.

The goal of this is twofold:
- Make it safe for contributors to be able to send large changes in increments, rather than monolithic monster commits, even if it temporarily means that `dev` is less stable. This will also help give the development community some visibility into what's actively being worked on.
- Make it less threatening for new AkkaDotNet contributors to send their first pull request, knowing that their contributions are going into a branch that is intended for receiving new changes.

`master` branch will remain the default for this project for when people clone or fork. Because AkkaDotNet is so new and many .NET developers haven't been exposed to Actor systems before, I thought it best that they be able to clone and run something that works well out of the box, versus something that might be less stable.

I added a [Contributing to Akka.NET](https://github.com/akkadotnet/akka.net/wiki/Contributing-to-Akka.NET) page which explains some of this and directs developers to send their new pull requests to the `dev` branch.

Any differing opinions or additional thoughts regarding this?
